### PR TITLE
[MAINTENANCE] Another docs build fix

### PR DESCRIPTION
**Pull request by tyler-hoffman**
This reverts commit 6bb0ea83b277094760432ee81511cddfcb313df5. It also updates `prepare_prior_versions.py` to be a bit more optimized so it doesn't time out in netlify. I'd rather do that as 2 PRs, but doing it together proves we get things totally green in CI.

this PR introduced the errors seen here https://github.com/great-expectations/great_expectations/actions/runs/7543744906/job/20535756842 and I'd like to isolate the various build issues.

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

**Review by cdkini — APPROVED** [review]

File: docs/prepare_prior_versions.py [hunk 1]
```diff
@@ -590,6 +590,7 @@ def prepare_prior_versions() -> None:
     update_tag_references_for_correct_version()
     use_relative_path_for_imports()
     prepend_version_info_to_name_for_md_relative_links_to_index_files()
+    prepend_version_info_to_name_for_md_relative_links()
```
Comment (tyler-hoffman) [inline docs/prepare_prior_versions.py:564]: Missed this in https://github.com/great-expectations/great_expectations/pull/9234 and to see the docs build go green, we need this change and all the removed public_api decorators.

File: docs/prepare_prior_versions.py [hunk 2]
```diff
@@ -55,7 +55,22 @@ def _paths_to_versioned_docs_after_v0_15_50() -> list[pathlib.Path]:
     paths = [
         f
         for f in data_path.iterdir()
-        if f.is_dir() and ("0.14.13" not in str(f) or "0.15.50" not in str(f))
```
Comment (tyler-hoffman) [inline docs/prepare_prior_versions.py:58]: the or here looks like it was just making this return True for every version AFAICT

Comment (cdkini) [reply docs/prepare_prior_versions.py:58]: What is the significance of these strings? Could we make them constants to give them some more context?

File: docs/prepare_prior_versions.py (hunk 1 above)
Comment (cdkini) [reply docs/prepare_prior_versions.py:564]: What's the plan to revert the reversion?

File: docs/prepare_prior_versions.py (hunk 2 above)
Comment (tyler-hoffman) [reply docs/prepare_prior_versions.py:58]: They're past versions. We could make them constants, but I think it'd still be fragile. What if we just plan to fix this in https://greatexpectations.atlassian.net/browse/V1-131?

File: docs/prepare_prior_versions.py (hunk 1 above)
Comment (tyler-hoffman) [reply docs/prepare_prior_versions.py:564]: The diff makes it hard to read, but `prepend_version_info_to_name_for_md_relative_links` and `prepend_version_info_to_name_for_md_relative_links_to_index_files` were merged together and now both are handled in `prepend_version_info_to_name_for_md_relative_links`.

**Pull request merged**